File: src/main/java/leetCode/t412/java/Solution.java
package leetCode.t412.java;

import java.util.ArrayList;
import java.util.List;

class Solution {
    public List<String> fizzBuzz(int n) {
        String fizz = "Fizz";
        String buzz = "Buzz";
        String fizzBuzz = "FizzBuzz";

        List<String> result = new ArrayList<>(n);
        for (int i = 1; i <= n; i++) {
            if (i % 3 == 0 && i % 5 == 0) {
                result.add(fizzBuzz);
            } else if (i % 3 == 0) {
                result.add(fizz);
            } else if (i % 5 == 0) {
                result.add(buzz);
            } else {
                result.add(String.valueOf(i));
            }
        }
        return result;
    }
}

File: src/main/java/leetCode/t412/kotlin/Solution.kt
package leetCode.t412.kotlin

class Solution {
    fun fizzBuzz(n: Int): List<String> {
        val fizz = "Fizz"
        val buzz = "Buzz"
        val fizzbuzz = "FizzBuzz"

        val result = mutableListOf<String>()
        for (i in 1..n) {
            if (i % 3 == 0 && i % 5 == 0) {
                result.add(fizzbuzz)
            } else if (i % 5 == 0) {
                result.add(buzz)
            } else if (i % 3 == 0) {
                result.add(fizz)
            } else {
                result.add(i.toString())
            }
        }
        return result
    }
}